arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -1703,6 +1584,110 @@ fun <A, B, C> Either<A, B>.zip(fb: Either<A, C>): Either<A, Pair<B, C>> =
     fb.map { b -> Pair(a, b) }
   }
 
+inline fun <A, B, C, D, E> Either<A, B>.zip(
+  c: Either<A, C>,
+  d: Either<A, D>,
+  map: (B, C, D) -> E
+): Either<A, E> =
+  zip(c, d, Either.unit, Either.unit, Either.unit, Either.unit, Either.unit, Either.unit, Either.unit) { b, c, d, _, _, _, _, _, _, _ -> map(b, c, d) }
+
+inline fun <A, B, C, D, E, F> Either<A, B>.zip(
+  c: Either<A, C>,
+  d: Either<A, D>,
+  e: Either<A, E>,
+  map: (B, C, D, E) -> F
+): Either<A, F> =
+  zip(c, d, e, Either.unit, Either.unit, Either.unit, Either.unit, Either.unit, Either.unit) { b, c, d, e, _, _, _, _, _, _ -> map(b, c, d, e) }
+
+inline fun <A, B, C, D, E, F, G> Either<A, B>.zip(
+  c: Either<A, C>,
+  d: Either<A, D>,
+  e: Either<A, E>,
+  f: Either<A, F>,
+  map: (B, C, D, E, F) -> G
+): Either<A, G> =
+  zip(c, d, e, f, Either.unit, Either.unit, Either.unit, Either.unit, Either.unit) { b, c, d, e, f, _, _, _, _, _ -> map(b, c, d, e, f) }
+
+inline fun <A, B, C, D, E, F, G, H> Either<A, B>.zip(
+  c: Either<A, C>,
+  d: Either<A, D>,
+  e: Either<A, E>,
+  f: Either<A, F>,
+  g: Either<A, G>,
+  map: (B, C, D, E, F, G) -> H
+): Either<A, H> =
+  zip(c, d, e, f, g, Either.unit, Either.unit, Either.unit, Either.unit) { b, c, d, e, f, g, _, _, _, _ -> map(b, c, d, e, f, g) }
+
+inline fun <A, B, C, D, E, F, G, H, I> Either<A, B>.zip(
+  c: Either<A, C>,
+  d: Either<A, D>,
+  e: Either<A, E>,
+  f: Either<A, F>,
+  g: Either<A, G>,
+  h: Either<A, H>,
+  map: (B, C, D, E, F, G, H) -> I
+): Either<A, I> =
+  zip(c, d, e, f, g, h, Either.unit, Either.unit, Either.unit) { b, c, d, e, f, g, h, _, _, _ -> map(b, c, d, e, f, g, h) }
+
+inline fun <A, B, C, D, E, F, G, H, I, J> Either<A, B>.zip(
+  c: Either<A, C>,
+  d: Either<A, D>,
+  e: Either<A, E>,
+  f: Either<A, F>,
+  g: Either<A, G>,
+  h: Either<A, H>,
+  i: Either<A, I>,
+  map: (B, C, D, E, F, G, H, I) -> J
+): Either<A, J> =
+  zip(c, d, e, f, g, h, i, Either.unit, Either.unit) { b, c, d, e, f, g, h, i, _, _ -> map(b, c, d, e, f, g, h, i) }
```

Comment:
just realized we have `Either.unit` vs `Eval.Unit`

Reply:
`Either.unit` is `internal` and `Eval.Unit` is deprecated ;)